litex_boards/community/platforms/marblemini.py

```diff
@@ -0,0 +1,253 @@
+from litex.build.generic_platform import *
+from litex.build.xilinx import XilinxPlatform, VivadoProgrammer
+
+# IOs ----------------------------------------------------------------------------------------------
+
+# TODO:
+# 1. TMDS lanes for the HDMI connector
+# 2. Populate more SFPs
+
+_io = [
+    ("clk20_vcxo", 0, Pins("D17"), IOStandard("LVCMOS33")),
+    ("vcxo_en", 0, Pins("E13"), IOStandard("LVCMOS33")),  # Enable for the above clock
+
+    ("mgt_clk_0", 0,
+     Subsignal("p", Pins("F6")),
+     Subsignal("n", Pins("E6"))),
+    ("mgt_clk_1", 0,
+     Subsignal("p", Pins("F10")),
+     Subsignal("n", Pins("E10"))),
+    ("serial", 0,
+        Subsignal("rts", Pins("W9")),
+        Subsignal("rx", Pins("U7")),
+        Subsignal("tx", Pins("Y9")),
+        IOStandard("LVCMOS25")
+    ),
+    ("eth_clocks", 0,
+     Subsignal("tx", Pins("J15"), Misc("SLEW=FAST")),
+     Subsignal("rx", Pins("L19")),
+     IOStandard("LVCMOS25")),
+    ("eth", 0, Subsignal("rx_ctl", Pins("H15"), Misc("IOB=TRUE")),
+     Subsignal("rx_data", Pins("K13 H14 J14 K14"), Misc("IOB=TRUE")),
+     Subsignal("tx_ctl", Pins("J16"), Misc("IOB=TRUE"), Misc("SLEW=FAST")),
+     Subsignal("tx_data", Pins("G15 G16 G13 H13"), Misc("SLEW=FAST")),
+     Subsignal("rst_n", Pins("M17")), IOStandard("LVCMOS25"),
+     ),
+    ("pmod", 0,
+     Subsignal("pmod", Pins("C18 D22 E22 G21 D21 E21 F21 G22"), IOStandard("LVCMOS33"))),
+    ("pmod", 1,
+     Subsignal("pmod", Pins("F13 C14 C15 D16 F14 F15 F16 E16"), IOStandard("LVCMOS33"))),
+    # ("sfp", 0,
+    #     Subsignal("txp", Pins("AC10")),
+    #     Subsignal("txn", Pins("AD10")),
+    #     Subsignal("rxp", Pins("AC12")),
+    #     Subsignal("rxn", Pins("AD12")),
+    # ),
+    ("ddram",
+        0,
+        Subsignal("a",
+                  Pins("L6 M5 P6 K6 M1 M3 N2 M6", "P1 P2 L4 N5 L3 R1 N3 E3"),
+                  IOStandard("SSTL135")),
+        Subsignal("ba", Pins("L5 M2 N4"), IOStandard("SSTL135")),
+        Subsignal("ras_n", Pins("J4"), IOStandard("SSTL135")),
+        Subsignal("cas_n", Pins("J6"), IOStandard("SSTL135")),
+        Subsignal("we_n", Pins("K3"),  IOStandard("SSTL135")),
+     # Subsignal("cs_n", Pins("T3"), IOStandard("SSTL135")),  # TODO: couldn't find chip select on Marble
```

Comment:
`cs_n` is indeed not mandatory, you can just keep this line commented and remove put an empty `Pins("")`.